modules/telemetry/src/metric_record.cpp

```diff
@@ -0,0 +1,152 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include "hephaestus/telemetry/metric_record.h"
+
+#include <cerrno>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <absl/base/thread_annotations.h>
+#include <absl/log/log.h>
+
+#include "hephaestus/concurrency/message_queue_consumer.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
+namespace heph::telemetry {
+
+namespace internal {
+namespace {
+[[nodiscard]] auto stringToInt64(const std::string& str) -> std::optional<int64_t> {
+  const char* start = str.c_str();
+  char* end{};
+  errno = 0;
+
+  static constexpr int BASE = 10;
+  int64_t result = std::strtoll(start, &end, BASE);
```

Comment:
IMO the error handling of https://en.cppreference.com/w/cpp/utility/from_chars is superior to strtoll and more modern